### ISSUE TITLE
.parse() result has empty-list (real) members

### ISSUE THREAD

**Issue by ThomasWaldmann** [matze/pkgconfig#40]
pkginfo 1.4.0
```
>>> from pkgconfig import parse
>>> parse('openssl')
defaultdict(<class 'list'>, {'libraries': ['ssl', 'crypto'], 'define_macros': []})
>>> parse('liblz4')
defaultdict(<class 'list'>, {'libraries': ['lz4'], 'define_macros': []})
>>> parse('libzstd')
defaultdict(<class 'list'>, {'libraries': ['zstd'], 'define_macros': []})
>>> parse('libxxxxxxxxxx')
defaultdict(<class 'list'>, {'define_macros': []})
```

Notable:
- define_macros always there, even if empty (other empty stuff not there and as it is a list-defaultdict, it does not need to be there when empty)
- for a non-existing lib (see last example), guess there could be some exception, but it fails silently.

**Comment by matze** [comment]
Good points, do you want to take care of that?

**Comment by ThomasWaldmann** [comment]
It's a defaultdict (with an empty-list default), so no burden: `parse_result['unknown_key'] == []`

I am working with `pkgconfig` to improve `borgbackup`'s library/header discovery and make the code simpler. It is still WIP, but you can see some results already there:

https://github.com/borgbackup/borg/pull/4437/files

Especially see the `members_appended()` helper and how it processes the `*_ext_kwargs()` call results (which might be directly the results of `pkgconfig.parse('somelib')` calls).

Also note how for the same library, some keys might be there or not (e.g. in my case, `define_macros` only is there if i want to use the bundled libs. in the other case it is not there and if no other lib added define_macros either, the defaultdict behaviour will result in an empty list when the Extension is built, as desired).

Also, when working based on bundled source code, `sources` key will be present and library related keys won't be. And vice versa when working based on a installed pre-compiled library.

So, if I am understanding correctly, I don't see an issue here.

**Comment by matze** [comment]
> It's a defaultdict (with an empty-list default), so no burden: `parse_result['unknown_key'] == []`

Dang, you are right :man_facepalming:

**Comment by ThomasWaldmann** [comment]
"for a non-existing lib (see last example), guess there could be some exception, but it fails silently."

moving this to a new issue.

so this issue is only about the empty-list members and can be closed after merging #41.

**Comment by matze** [comment]
Next time just prefix with "Fix #41 …" and the ticket should be auto-closed.

**Comment by ThomasWaldmann** [comment]
I usually append ", fixes #NNN", but at the time of doing the commit, it was no complete fix, but just a partial one and thus just a "see #NNN" comment.
